drivers/digital-io/max22190/iio_max22190.c

```diff
@@ -0,0 +1,608 @@
+				    const struct iio_ch_info *channel,
+				    intptr_t priv);
+
+static int max22190_iio_write_filter(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv);
+
+static int max22190_iio_read_filter_delay(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int max22190_iio_write_filter_delay(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int max22190_iio_read_filter_available(void *dev, char *buf,
+		uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int max22190_iio_read_fault1(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv);
+
+static int max22190_iio_read_fault2(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv);
+
+static int max22190_iio_reg_read(struct max22190_iio_desc *dev, uint32_t reg,
+				 uint32_t *readval);
+
+static int max22190_iio_reg_write(struct max22190_iio_desc *dev, uint32_t reg,
+				  uint32_t writeval);
+
+static int max22190_iio_setup_channels(struct max22190_iio_desc *desc,
+				       bool *ch_enabled);
+
+static const uint32_t max22190_delay_avail[8] = {
+	50, 100, 400, 800, 1800, 3200, 12800, 20000
+};
+
+static struct iio_attribute max22190_attrs[] = {
+	{
+		.name = "raw",
+		.show = max22190_iio_read_raw,
+	},
+	{
+		.name = "offset",
+		.show = max22190_iio_read_offset,
+	},
+	{
+		.name = "scale",
+		.show = max22190_iio_read_scale,
+	},
+	{
+		.name = "filter_bypass",
+		.show = max22190_iio_read_filter,
+		.store = max22190_iio_write_filter,
+	},
+	{
+		.name = "filter_delay",
+		.show = max22190_iio_read_filter_delay,
+		.store = max22190_iio_write_filter_delay,
+	},
+	{
+		.name = "filter_delay_available",
+		.show = max22190_iio_read_filter_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "fault1",
+		.show = max22190_iio_read_fault1,
+	},
+	{
+		.name = "fault2",
+		.show = max22190_iio_read_fault2,
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_device max22190_iio_dev = {
+	.debug_reg_read = (int32_t (*)())max22190_iio_reg_read,
+	.debug_reg_write = (int32_t (*)())max22190_iio_reg_write,
+};
+
+/**
+ * @brief Read the raw attribute for a specific channel
+ *
+ * @param dev - The iio device structure
+ *
+ * @param buf - Buffer to be filled with requested data.
+ *
+ * @param len - Length of the received command buffer in bytes.
+ *
+ * @param channel - Command channel info.
+ *
+ * @param priv - Private descriptor.
+ *
+ * @return 0 in case of succes, error code otherwise.
+*/
+static int max22190_iio_read_raw(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	struct max22190_iio_desc *desc = dev;
+	uint32_t val;
+	int ret;
+
+	ret = max22190_reg_read(desc->max22190_desc, MAX22190_DIGITAL_INPUT_REG,
```

Comment:
This is not for a specific channel, you're just storing the value of a register.

Reply:
Will use no_os_field_get, my bad.